File: weyl-core/src/main/java/io/quartic/weyl/core/model/Layer.java
package io.quartic.weyl.core.model;

import com.vividsolutions.jts.index.SpatialIndex;
import io.quartic.common.SweetStyle;
import io.quartic.weyl.core.feature.FeatureCollection;
import org.immutables.value.Value;

import java.util.Collection;


@SweetStyle
@Value.Immutable
public interface Layer {
    LayerSpec spec();
    FeatureCollection features();

    // Index features
    SpatialIndex spatialIndex();
    Collection<IndexedFeature> indexedFeatures();
    LayerStats stats();
}

File: weyl-core/src/main/kotlin/io/quartic/weyl/core/model/Layer.kt
package io.quartic.weyl.core.model

import com.vividsolutions.jts.index.SpatialIndex
import io.quartic.weyl.core.feature.FeatureCollection

data class Layer(
        val spec: LayerSpec,    // Should be able to remove this
        val features: FeatureCollection,
        val dynamicSchema: DynamicSchema,

        // Index features
        val spatialIndex: SpatialIndex,

        val indexedFeatures: Collection<IndexedFeature>,
        val stats: LayerStats
)
